katharsis-core/src/main/java/io/katharsis/queryspec/InMemoryEvaluator.java

```diff
@@ -97,11 +97,15 @@ private static boolean matchesPrimitiveOperator(Object object, FilterSpec filter
 		Object value = PropertyUtils.getProperty(object, filterSpec.getAttributePath());
 		FilterOperator operator = filterSpec.getOperator();
 		Object filterValue = filterSpec.getValue();
-		if (value instanceof Collection) {
-			return matchesAny((Collection<?>) value, operator, filterValue);
-		}
-		else {
-			return operator.matches(value, filterValue);
+		if (filterValue != null ||  (filterValue == null && (operator.equals(FilterOperator.EQ) || operator.equals(FilterOperator.NEQ)))) {
```

Comment:
I am looking at [this line](https://github.com/apetrucci/katharsis-framework/blob/master/pom.xml#L107) and [this line](https://github.com/apetrucci/katharsis-framework/blob/master/pom.xml#L20) and it is ready to go with Java 8 - I am pretty sure we cleaned up any documentation that said Java 7. You're clear to use lambas

Did I miss something